Unity-Technologies/ScriptableRenderPipeline
# Fixes for terrain heightmap default/null values and epsilons

### Purpose of this PR
Fix for https://fogbugz.unity3d.com/f/cases/1187428/
For terrain that doesn't use all available layers, empty layers have their masks initialized to maximum height. This causes artifacting in the remaining layers. 

---
### Testing status

**Manual Tests**: What did you do?
- Other: 
When I ran graphics tests locally, the only tests that failed were ones that also failed on HDRP/staging.

Visually confirmed that scaling height transition causes smoothly changing differences in appearance. 
In URP, for very small values of height transition, it seems like the edges of the splat map aren't smoothed out correctly, but I was unable to find a version of URP where this doesn't happen:
![image](https://user-images.githubusercontent.com/3450690/66001392-3d672100-e456-11e9-8a67-d2bd2922c58b.png)

In HDRP, the edges of the splatmap are smooth as expected.

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)
None, but maybe we should have a test for height transition?

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: Low - single line change in HDRP, only initial value change in URP

**Halo Effect**: None

---
### Comments to reviewers
Landon, could you please test this branch?
